src/bootman/sysconfig.c

```diff
@@ -237,10 +175,39 @@ SystemConfig *cbm_inspect_root(const char *path, bool image_mode)
         c->prefix = realp;
         c->wanted_boot_mask = 0;
 
-        if (image_mode) {
-                cmb_inspect_root_image(c, realp);
+        if (getenv("CBM_FORCE_LEGACY")) {
+                force_legacy = true;
+        }
+
+        legacy_boot = get_legacy_boot_device(realp);
+        uefi_boot = get_boot_device();
+
+        /* typically /sys, but we forcibly fail this with our tests */
+        fw_path = string_printf("%s/firmware/efi", cbm_system_get_sysfs_path());
+        native_uefi = nc_file_exists(fw_path);
+
+        /*
+         * uefi has precedence over legacy, if we detected both uefi wins
+         * but if force_legacy is set then we honor "users choice"
+         */
+        if (!force_legacy && ((legacy_boot && uefi_boot) || uefi_boot)) {
+                c->wanted_boot_mask = BOOTLOADER_CAP_UEFI;
+        } else if (!native_uefi && (legacy_boot || force_legacy)) {
+                c->wanted_boot_mask = BOOTLOADER_CAP_LEGACY;
+        } else if (native_uefi && !force_legacy) {
+                c->wanted_boot_mask = BOOTLOADER_CAP_UEFI;
         } else {
-                cmb_inspect_root_native(c, realp);
+                c->wanted_boot_mask = BOOTLOADER_CAP_LEGACY;
+        }
+
+        if (c->wanted_boot_mask != 0) {
```

Comment:
I think this condition can be removed now since we always will have set either legacy or uefi now.